From efafcf808463247ec361bb8de31d3086a43fffbe Mon Sep 17 00:00:00 2001
From: Matthew Waters <matthew@centricular.com>
Date: Mon, 5 Nov 2018 22:49:46 +1100
Subject: [PATCH] build/meson: fix compilation on arm64

The assembly files used don't use the right comments for arm64
---
 meson.build | 20 ++++++++++++--------
 1 file changed, 12 insertions(+), 8 deletions(-)

diff --git a/meson.build b/meson.build
index 83fb62c..80242d6 100644
--- a/meson.build
+++ b/meson.build
@@ -46,21 +46,25 @@ have_arm = false
 have_armv7 = false
 have_neon = false
 have_x86 = false
-if ['arm', 'armv7', 'aarch64'].contains(host_machine.cpu_family())
-  have_arm = true
-  arch_cflags = ['-DWEBRTC_ARCH_ARM']
+if ['arm', 'armv7'].contains(host_machine.cpu_family())
+  if cc.compiles('''#ifdef __ARM_ARCH_ISA_ARM
+#error no arm arch
+#endif''')
+    have_arm = true
+    arch_cflags += ['-DWEBRTC_ARCH_ARM']
+  endif
   if cc.compiles('''#ifndef __ARM_ARCH_7A__
 #error no armv7 arch
 #endif''')
     have_armv7 = true
-    arch_cflags = ['-DWEBRTC_ARCH_ARM_V7']
+    arch_cflags += ['-DWEBRTC_ARCH_ARM_V7']
   endif
-  if cc.compiles('''#ifndef __aarch64__
+endif
+if cc.compiles('''#ifndef __aarch64__
 #error no aarch64 arch
 #endif''')
-    have_neon = true
-    arch_cflags = ['-DWEBRTC_ARCH_ARM64', '-DWEBRTC_HAS_NEON']
-  endif
+  have_neon = true
+  arch_cflags += ['-DWEBRTC_ARCH_ARM64', '-DWEBRTC_HAS_NEON']
 endif
 if ['x86', 'x86_64'].contains(host_machine.cpu_family())
   have_x86 = true
-- 
2.19.0

